packages/app/src/app/components/SubscribeForm/CheckoutForm/index.tsx

```diff
@@ -1,111 +1,104 @@
-import React from 'react';
-import { injectStripe, CardElement } from 'react-stripe-elements';
-import { Button } from '@codesandbox/common/lib/components/Button';
+import React, { useState } from 'react';
+import { injectStripe } from 'react-stripe-elements';
 import { logError } from '@codesandbox/common/lib/utils/analytics';
+import {
+  CardContainer,
+  Card,
+  NameInput,
+  ErrorText,
+  Label,
+  Submit,
+} from './elements';
+
+interface ICheckoutFormProps {
+  name?: string;
+  loading: boolean;
+  buttonName: string;
+  loadingText: string;
+  error: Error;
+  stripe: any;
+  subscribe: (id: string) => void;
+}
 
-import { CardContainer, NameInput, ErrorText, Label } from './elements';
-
-class CheckoutForm extends React.PureComponent {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      errors: {},
-      name: props.name || '',
-    };
-  }
-
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.name !== this.props.name) {
-      this.setState({ errors: {}, name: nextProps.name });
-    }
-  }
-
-  setName = e => {
-    if (e) {
-      this.setState({ errors: {}, name: e.target.value });
-    }
-  };
+interface IErrorsState {
```

Comment:
@Saeris I think you should use a normal `type` instead of an `interface`

```suggestion
type ErrorsState = {
```